core/src/gl/renderState.cpp

```diff
@@ -65,6 +72,14 @@ namespace RenderState {
         textureUnit.init(std::numeric_limits<unsigned int>::max(), false);
     }
 
+    bool isCurrentGeneration(int _generation) {
```

Comment:
I think we can use the same generation process through RenderState for Texture and Shaders.